project_work_time_control/model/account_analytic_line.py

```diff
@@ -25,38 +25,48 @@
 from datetime import datetime
 
 
-class ProjectWork(models.Model):
-    _inherit = 'project.task.work'
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
 
-    project = fields.Many2one(
-        string='Project', related='task_id.project_id', store=True,
-        domain="[('state', 'not in', ('template', 'cancelled', 'close'))]")
+    date_time = fields.Datetime(default=fields.Datetime.now, string='Date')
     include_in_task_work_view = fields.Boolean(
         related='task_id.stage_id.include_in_task_work_view', readonly=True,
         store=True)
 
-    @api.onchange('project')
-    def onchange_project(self):
-        if not self.project:
+    @api.onchange('account_id')
+    def onchange_account_id(self):
+        if not self.account_id:
             return {'domain': {'task_id': []}}
+        self.task_id = False
+        project = self.env['project.project'].search(
+            [('analytic_account_id', '=', self.account_id.id)], limit=1)
         return {
             'domain': {
-                'task_id': [('project_id', '=', self.project.id),
+                'task_id': [('project_id', '=', project.id),
```

Comment:
Well, it's not exactly the same, because you have 2 fields for this. I think that if you archive it, you won't see it more (active field behavior) unless you make the corresponding advanced search. In contrast, task state reflects where in your pipeline it's.